groups/nts/ntsu/ntsu_socketutil.cpp

```diff
@@ -539,34 +542,180 @@ void IncomingDataStats::update(const struct iovec* buffersReceivable,
 
 #endif
 
+#if defined(BSLS_PLATFORM_OS_UNIX)
+
+// Provide a buffer suitable for attaching to a call to 'sendmsg' to send
+// control data to the peer of a socket.
+class SendControl 
+{
+    enum {
+        // The payload size required to send any meta-data (viz. open file 
+        // descriptors) to the peer of a socket.
+        k_SEND_CONTROL_PAYLOAD_SIZE = static_cast<int>(
+            NTSU_SOCKETUTIL_MAX_HANDLES_PER_OUTGOING_CONTROLMSG * sizeof(int)),
+
+        // The control buffer capacity required to send any meta-data (viz.
+        // open file descriptors) to the peer of a socket.
+        k_SEND_CONTROL_BUFFER_SIZE = 
+            static_cast<int>(CMSG_SPACE(k_SEND_CONTROL_PAYLOAD_SIZE))
+    };
+
+    // Define a type alias for a maximimally-aligned buffer of suitable size to
+    // send any meta-data (viz. open file descriptors) to the peer of the 
+    // socket.
+    typedef bsls::AlignedBuffer<k_SEND_CONTROL_BUFFER_SIZE> Arena;
+
+    Arena d_arena;
+
+  private:
+    SendControl(const SendControl&) BSLS_KEYWORD_DELETED;
+    SendControl& operator=(const SendControl) BSLS_KEYWORD_DELETED;
+
+  public:
+    // Create a new send control buffer.
+    SendControl();
+        
+    // Destroy this object.
+    ~SendControl();
+        
+    // Encode the specified 'options' into the control buffer. Return the 
+    // error.
+    ntsa::Error encode(msghdr* msg, const ntsa::SendOptions& options);
+};
+
+// Provide a buffer suitable for attaching to a call to 'recvmsg' to receive
+// control data from the peer of a socket.
+class ReceiveControl 
+{
+    enum {
+        // The payload size required to receive any meta-data  (e.g. open 
+        // file descriptors, timestamps, etc.) buffered by the operating system
+        // for a socket.
+        k_RECEIVE_CONTROL_PAYLOAD_SIZE = static_cast<int>(
+            NTSU_SOCKETUTIL_MAX_HANDLES_PER_INCOMING_CONTROLMSG * sizeof(int))
```

Comment:
`sizeof(int)`  is everywhere to mean amount of memory required to represent transferred file handle. I suggest to either use an explicit constant instead of this `sizeof` or to define a new typedef and use it instead of just `int`.

Reply:
No Unix that I know of defines a typedef for the type of a file descriptor, it's just an `int`. However, we typedef `ntsa::Handle` to `int` on Unixes. I have revised most places of the code to use `ntsa::Handle` instead of `int` and either use `sizeof <variable>` or `sizeof(ntsa::Handle)`.